### ISSUE TITLE
Faster RCNN: Support Batch Sizes larger than One per GPU

### ISSUE THREAD

**Issue by jwiens0** [tensorpack/tensorpack#1227]
During training, the Faster R-CNN example does not support a batch size larger than one per GPU.

**Comment by ppwwyyxx** [comment]
You're right.

As said in the [issue template](https://github.com/tensorpack/tensorpack/issues/new?template=feature-requests.md):

> + "Could you improve/implement an example/paper ?"
  -- The answer is: we have no plans to do so. We don't consider feature
  requests for examples or implement a paper for you, unless it demonstrates 
  some Tensorpack features not yet demonstrated in the existing examples.
  If you don't know how to do something yourself, you may ask a usage question.

**Comment by jwiens0** [comment]
This is a feature that someone could add easily? I am surprised that I couldn't find some hard-coded variable to change the batch size. Or, is this a limitation of tensorpack or the implementation of Faster R-CNN.

**Comment by ppwwyyxx** [comment]
This is not a feature that someone could add easily. 
And as mentioned in https://github.com/tensorpack/tensorpack/blob/master/examples/FasterRCNN/NOTES.md this is a limitation of tensorflow (or a limitation of tensor-based static graph computation framework, in general).

**Comment by armandmcqueen** [comment]
We have an implementation based on Tensorpack's that supports larger batch sizes [here](https://github.com/armandmcqueen/tensorpack-mask-rcnn).

**Comment by jwiens0** [comment]
Thanks, I will take a look.
